### ISSUE TITLE
Jottacloud authentication failing with 'cannot fetch token: 400 Bad Request'

### ISSUE THREAD

**Issue by gerph** [rclone/rclone#4645]
#### What is the problem you are having with rclone?

Using the backend Jottacloud, with the personal token for login, it works for about an hour or so and then starts failing with an error saying it cannot fetch token.

#### What is your rclone version (output from `rclone version`)

Installed from brew:

```
charles@laputa ~/backup (master) [1]> rclone version
rclone v1.53.1-DEV
- os/arch: darwin/amd64
- go version: go1.15.2
```

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)

MacOS 10.14.6

####  Which cloud storage system are you using? (eg Google Drive)

Jottacloud, with cache and crypt.

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)

`rclone copy file jottacloud:...`

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)

At the point at which it started happening I only have a log of it with `-v`:

2020/10/01 23:30:31 ERROR : Chris Wilkins/The Story Of The Commodore Amiga In Pixels (5175)/The Story Of The Commodore Amiga In Pixels - Chris Wilkins.pdf: put: error uploading: Post "https://api.jottacloud.com/files/v1/allocate": couldn't fetch token - maybe it has expired? - refresh with "rclone config reconnect jottacloud:": oauth2: cannot fetch token: 400 Bad Request

Repeating the command to get it to fail:

```
rclone -vv --checkers 10 --transfers 10 --bwlimit 1200 copy '/Users/charles/Calibre Library' jottacloud-cache-crypt:archives/2020-10/laputa/calibre 2>&1 | tee log
2020/10/01 23:44:37 DEBUG : rclone: Version "v1.53.1-DEV" starting with parameters ["rclone" "-vv" "--checkers" "10" "--transfers" "10" "--bwlimit" "1200" "copy" "/Users/charles/Calibre Library" "jottacloud-cache-crypt:archives/2020-10/laputa/calibre"]
2020/10/01 23:44:37 DEBUG : Creating backend with remote "/Users/charles/Calibre Library"
2020/10/01 23:44:37 DEBUG : Using config file from "/Users/charles/.config/rclone/rclone.conf"
2020/10/01 23:44:37 INFO  : Starting bandwidth limiter at 1.172MBytes/s
2020/10/01 23:44:37 DEBUG : Creating backend with remote "jottacloud-cache-crypt:archives/2020-10/laputa/calibre"
2020/10/01 23:44:37 DEBUG : Creating backend with remote "jottacloud-cache:/data/i1ge5ig5e0ukqu3cr6u2h39rj8/n65i27huls30nbk7hlqe1da44g/nshgk8tnqg5mfrbrbn0c5am3l4/db0v7metvbjhkj1269aprjfr5k"
2020/10/01 23:44:37 DEBUG : Creating backend with remote "jottacloud:/data/i1ge5ig5e0ukqu3cr6u2h39rj8/n65i27huls30nbk7hlqe1da44g/nshgk8tnqg5mfrbrbn0c5am3l4/db0v7metvbjhkj1269aprjfr5k"
2020/10/01 23:44:37 DEBUG : jottacloud root 'data/i1ge5ig5e0ukqu3cr6u2h39rj8/n65i27huls30nbk7hlqe1da44g/nshgk8tnqg5mfrbrbn0c5am3l4/db0v7metvbjhkj1269aprjfr5k': Token expired but no uploads in progress - doing nothing
2020/10/01 23:44:37 DEBUG : jottacloud: Loaded invalid token from config file - ignoring
2020/10/01 23:44:37 DEBUG : jottacloud: Token refresh failed try 1/5: oauth2: cannot fetch token: 400 Bad Request
Response: {"error":"unauthorized_client","error_description":"INVALID_CREDENTIALS: Invalid client credentials"}
2020/10/01 23:44:38 DEBUG : jottacloud: Loaded invalid token from config file - ignoring
2020/10/01 23:44:38 DEBUG : jottacloud: Token refresh failed try 2/5: oauth2: cannot fetch token: 400 Bad Request
Response: {"error":"unauthorized_client","error_description":"INVALID_CREDENTIALS: Invalid client credentials"}
2020/10/01 23:44:39 DEBUG : jottacloud: Loaded invalid token from config file - ignoring
2020/10/01 23:44:40 DEBUG : jottacloud: Token refresh failed try 3/5: oauth2: cannot fetch token: 400 Bad Request
Response: {"error":"unauthorized_client","error_description":"INVALID_CREDENTIALS: Invalid client credentials"}
2020/10/01 23:44:41 DEBUG : jottacloud: Loaded invalid token from config file - ignoring
2020/10/01 23:44:41 DEBUG : jottacloud: Token refresh failed try 4/5: oauth2: cannot fetch token: 400 Bad Request
Response: {"error":"unauthorized_client","error_description":"INVALID_CREDENTIALS: Invalid client credentials"}
2020/10/01 23:44:42 DEBUG : jottacloud: Loaded invalid token from config file - ignoring
2020/10/01 23:44:42 DEBUG : jottacloud: Token refresh failed try 5/5: oauth2: cannot fetch token: 400 Bad Request
Response: {"error":"unauthorized_client","error_description":"INVALID_CREDENTIALS: Invalid client credentials"}
2020/10/01 23:44:43 DEBUG : Creating backend with remote "jottacloud-cache:/data/i1ge5ig5e0ukqu3cr6u2h39rj8/n65i27huls30nbk7hlqe1da44g/nshgk8tnqg5mfrbrbn0c5am3l4/db0v7metvbjhkj1269aprjfr5k"
2020/10/01 23:44:43 DEBUG : Creating backend with remote "jottacloud:/data/i1ge5ig5e0ukqu3cr6u2h39rj8/n65i27huls30nbk7hlqe1da44g/nshgk8tnqg5mfrbrbn0c5am3l4/db0v7metvbjhkj1269aprjfr5k"
2020/10/01 23:44:43 DEBUG : jottacloud: Loaded invalid token from config file - ignoring
2020/10/01 23:44:43 DEBUG : jottacloud: Token refresh failed try 1/5: oauth2: cannot fetch token: 400 Bad Request
Response: {"error":"unauthorized_client","error_description":"INVALID_CREDENTIALS: Invalid client credentials"}
2020/10/01 23:44:44 DEBUG : jottacloud: Loaded invalid token from config file - ignoring
2020/10/01 23:44:44 DEBUG : jottacloud: Token refresh failed try 2/5: oauth2: cannot fetch token: 400 Bad Request
Response: {"error":"unauthorized_client","error_description":"INVALID_CREDENTIALS: Invalid client credentials"}
2020/10/01 23:44:45 DEBUG : jottacloud: Loaded invalid token from config file - ignoring
2020/10/01 23:44:45 DEBUG : jottacloud: Token refresh failed try 3/5: oauth2: cannot fetch token: 400 Bad Request
Response: {"error":"unauthorized_client","error_description":"INVALID_CREDENTIALS: Invalid client credentials"}
2020/10/01 23:44:46 DEBUG : jottacloud: Loaded invalid token from config file - ignoring
2020/10/01 23:44:46 DEBUG : jottacloud: Token refresh failed try 4/5: oauth2: cannot fetch token: 400 Bad Request
Response: {"error":"unauthorized_client","error_description":"INVALID_CREDENTIALS: Invalid client credentials"}
2020/10/01 23:44:47 DEBUG : jottacloud: Loaded invalid token from config file - ignoring
2020/10/01 23:44:47 DEBUG : jottacloud: Token refresh failed try 5/5: oauth2: cannot fetch token: 400 Bad Request
Response: {"error":"unauthorized_client","error_description":"INVALID_CREDENTIALS: Invalid client credentials"}
2020/10/01 23:44:48 DEBUG : jottacloud root 'data/i1ge5ig5e0ukqu3cr6u2h39rj8/n65i27huls30nbk7hlqe1da44g/nshgk8tnqg5mfrbrbn0c5am3l4/db0v7metvbjhkj1269aprjfr5k': Token expired but no uploads in progress - doing nothing
2020/10/01 23:44:48 Failed to create file system for "jottacloud-cache-crypt:archives/2020-10/laputa/calibre": failed to make remote "jottacloud-cache:/data/" to wrap: failed to make remote "jottacloud:/data/i1ge5ig5e0ukqu3cr6u2h39rj8/n65i27huls30nbk7hlqe1da44g/nshgk8tnqg5mfrbrbn0c5am3l4/db0v7metvbjhkj1269aprjfr5k" to wrap: couldn't get customer info: Get "https://api.jottacloud.com/account/v1/customer": couldn't fetch token - maybe it has expired? - refresh with "rclone config reconnect jottacloud:": oauth2: cannot fetch token: 400 Bad Request
Response: {"error":"unauthorized_client","error_description":"INVALID_CREDENTIALS: Invalid client credentials"}
```

**Comment by gerph** [comment]
I have reconfigured with a new token and installed the beta to see if the issue recurs.

**Comment by buengese** [comment]
That sounds like a problem with refreshing the authorization token. We had these a couple of times with Jottacloud in the past because they keep changing their authentication system. I couldn't reproduce this right now but let me now if it happens again.

**Comment by gerph** [comment]
Yeah, it's happened about an hour after I register the new personal token, with the beta as well:

```
charles@laputa ~/backup (master)> rclone version
rclone v1.54.0-beta.4793.2b7994e73
- os/arch: darwin/amd64
- go version: go1.15.2
```

**Comment by gerph** [comment]
I've had to reconfigure using the personal token about 5 times today - it's always failing with that error. In case it was an issue with the IP mode that meant that your test was different to me, I tried forcing to IPv4 by using `--bind <ipv4 address>` and still get the same error.

**Comment by buengese** [comment]
For some reason this problem doesn't seem to happen with my accounts. Does that error only occur during long transfers? Could you maybe provide with a log with the options `-vv --dump=bodies` (this produces a lot of output so the best way to this is to wait for the error to happen and then run a simple command like rclone about with these options). Note that this log includes all your auth tokens so you'll have to manually censor them.

As workaround you can try using legacy authentification though that's incompatible with 2 factor authentication.

**Comment by gerph** [comment]
I hope I've not wasted your time - I was just about to post the results of more investigation. I recreated the configuration from scratch and it all worked fine for over an hour, so I looked at what the difference was in my configuration file.

In the working version, I don't have a `client_id` or `client_secret` set, but in the failing one I do. I don't know what those are - I assume that they refer to the legacy configuration, and at some time in the past when I was experimenting with jottacloud (probably a couple of years ago), and proving to myself that rclone worked with it, I had set the legacy authentication. In all my reconfigurations this time around I had not selected to use the legacy authentication, and was oblivious to those settings.

IF that's the case, then it might be handy to clear out the legacy authentication when the personal token is used, so that it isn't used for the reauthentication process.

Anyhow, having cleared out those two variables from the configuration manually, has fixed the problem for me.
